guides/source/active_record_callbacks.md

```diff
@@ -170,21 +390,77 @@ WARNING. `after_save` runs both on create and update, but always _after_ the mor
+[`after_destroy`]:
+    https://api.rubyonrails.org/classes/ActiveRecord/Callbacks/ClassMethods.html#method-i-after_destroy
+[`around_destroy`]:
+    https://api.rubyonrails.org/classes/ActiveRecord/Callbacks/ClassMethods.html#method-i-around_destroy
+[`before_destroy`]:
+    https://api.rubyonrails.org/classes/ActiveRecord/Callbacks/ClassMethods.html#method-i-before_destroy
 
-NOTE: `before_destroy` callbacks should be placed before `dependent: :destroy` associations (or use the `prepend: true` option), to ensure they execute before the records are deleted by `dependent: :destroy`.
+NOTE: `before_destroy` callbacks should be placed before `dependent: :destroy`
+associations (or use the `prepend: true` option), to ensure they execute before
+the records are deleted by `dependent: :destroy`.
 
-WARNING. `after_commit` makes very different guarantees than `after_save`, `after_update`, and `after_destroy`. For example if an exception occurs in an `after_save` the transaction will be rolled back and the data will not be persisted. While anything that happens `after_commit` can guarantee the transaction has already completed and the data was persisted to the database. More on [transactional callbacks](#transaction-callbacks) below.
+WARNING. `after_commit` makes very different guarantees than `after_save`,
+`after_update`, and `after_destroy`. For example, if an exception occurs in an
+`after_save` the transaction will be rolled back and the data will not be
+persisted. However, anything that happens `after_commit` can guarantee the
+transaction has already been completed and the data was persisted to the
+database. More on [transactional callbacks](#transaction-callbacks) below.
+
+```ruby
+class User < ApplicationRecord
+  before_destroy :check_admin_count
+  around_destroy :log_destroy_operation
+  after_destroy :notify_users
+
+  private
+
+  def check_admin_count
+    if User.where(admin: true).count == 1 && admin?
+      raise StandardError.new("Cannot delete the last admin user")
+    end
+    Rails.logger.info("Checked the admin count")
+  end
+
+  def log_destroy_operation
+    Rails.logger.info("About to destroy user with ID #{id}")
+    yield
+    Rails.logger.info("User with ID #{id} destroyed successfully")
+  end
+
+  def notify_users
+    Rails.logger.info("Notification sent to other users about user deletion")
+  end
+end
+```
+
+```irb
+irb> user = User.find(1)
+=> #<User id: 1, email: "john.doe@gmail.com", created_at: "2024-03-20 16:19:52.405195000 +0000", updated_at: "2024-03-20 16:19:52.405195000 +0000", name: "John Doe", role: "admin" >
+
+irb> user.destroy
+Checked the admin count
+About to destroy user with ID 1
+User with ID 1 destroyed successfully
+Notification sent to other users about user deletion
+```
+
+`after_commit` / `after_rollback` examples can be found
+[here](active_record_callbacks.html#after-commit-and-after-rollback).
```

Comment:
Should we add a similar note to the create callbacks section? It's shown there, we have examples of other callback types, but don't mention these.

Reply:
✅